### ISSUE TITLE
mcfly.bash leaves behind the /tmp/ temporary files

### ISSUE THREAD

**Issue by grinapo** [cantino/mcfly#71]
$ ls -la /tmp/mcfly.* |wc -l
    130

**Comment by cantino** [comment]
`/tmp` should get cleaned up on restart.

**Comment by grinapo** [comment]
We talk about _large_ amounts of tmp files, in the magnitude of thousands.

**Comment by cantino** [comment]
Huh, that's odd. Are you opening and closing many terminal windows?

**Comment by grinapo** [comment]
Ermm... yes? :-)

    $ ps awuxf | grep xterm|wc -l
    64

**Comment by cantino** [comment]
Hmm, if you close terminals with `exit`, we could do this:

```bash
function mcfly_cleanup {
  if [ ! -z "${MCFLY_HISTORY}" ] && [[ -f "${MCFLY_HISTORY}" ]]; then
    rm "${MCFLY_HISTORY}"
  fi
}

trap mcfly_cleanup EXIT
```

If you use the UI, though, I'm not sure how to hook into cleanup. On my Mac, closing a Terminal doesn't seem to run `.bash_logout`. What about on your system?

**Comment by grinapo** [comment]
Nope, there is no sign of it in strace, either. 
`trap` works well, and I guess it works in most (all?) shells, not just bash.

**Comment by hegerj** [comment]
Currently I have 14 `mcfly.*` files in `/tmp`. All of them have today's timestamp. It's first time I'm noticing it but doesn't mean much as I do not check `/tmp` that often :) .
Using McFly 0.3.5.

**Comment by cantino** [comment]
These should get cleaned up when your system is restarted, right?

**Comment by hegerj** [comment]
They did but immediately after booting there are bunch of new ones.

**Comment by cantino** [comment]
Are they all identical?

**Comment by hegerj** [comment]
no, each is different (not sure about each, I compared random few which were different)

**Comment by waldner** [comment]
It would be nice to include the cleanup function and trap in the bash code generated by `mcfly init bash`.
